Adaptors/MongoDB/src/Options/MongoDB.cs

```diff
@@ -23,46 +23,109 @@
 
 namespace ArmoniK.Core.Adapters.MongoDB.Options;
 
+/// <summary>
+///   Represents the configuration settings for connecting to a MongoDB database.
+/// </summary>
 [PublicAPI]
 public class MongoDB
 {
+  /// <summary>
+  ///   The name of the configuration section for MongoDB settings.
+  /// </summary>
   public const string SettingSection = nameof(MongoDB);
 
+  /// <summary>
+  ///   Whether to use TLS for the connection.
+  /// </summary>
   public bool Tls { get; set; }
 
+  /// <summary>
+  ///   Whether to allow insecure TLS connections.
+  /// </summary>
   public bool AllowInsecureTls { get; set; }
 
+  /// <summary>
+  ///   Whether to connect directly to the MongoDB server, bypassing the replica set.
+  /// </summary>
   public bool DirectConnection { get; set; }
 
+  /// <summary>
+  ///   Name of the replica set to connect to.
+  /// </summary>
   public string ReplicaSet { get; set; } = "";
 
+  /// <summary>
+  ///   Hostname of the MongoDB server.
+  /// </summary>
   public string Host { get; set; } = "";
 
+  /// <summary>
+  ///   Port number for the MongoDB server connection.
+  /// </summary>
   public int Port { get; set; }
 
+  /// <summary>
+  ///   Path to the Certificate Authority (CA) file for TLS connections.
+  /// </summary>
   public string CAFile { get; set; } = "";
 
+  /// <summary>
+  ///   Path to the credentials file for authentication.
+  /// </summary>
   public string CredentialsPath { get; set; } = "";
 
+  /// <summary>
+  ///   Username for connecting to the MongoDB server.
+  /// </summary>
   public string User { get; set; } = "";
 
+  /// <summary>
+  ///   Connection string for the MongoDB server.
```

Comment:
You should add that it overrides part of the other connection options. You should take a look at the code to write something coherent